### DISCUSSION TITLE
Empty fileset should produce an error

### DISCUSSION THREAD

**Discussion by mgenereu** [terraform-linters/tflint#1825] [Ideas]
## Introduction

While running in CI/CD, `tflint` could be accidentally ran against the wrong directory.  If the wrong directory contains no Terraform files, this returns a success.  Success in CI/CD is rolled up into a green check mark and isn't caught right away.

## Proposal

`tflint` should produce an error if no files match.  Possibly with a command line flag if backwards compatibility is needed.

I would be happy to write a PR if this proposal makes sense.

**Comment by bendrucker** [comment]
Given that `terraform validate` passes on an empty directory I don't think we'd move forward with this. 

```sh
$ terraform validate
Success! The configuration is valid.
```

```sh
$ terraform plan
╷
│ Error: No configuration files
│
│ Plan requires configuration to be present. Planning without a configuration
│ would mark everything for destruction, which is normally not what is desired.
│ If you would like to destroy everything, run plan with the -destroy option.
│ Otherwise, create a Terraform configuration file (.tf file) and try again.
```

**Comment by mgenereu** [comment]
During the CI/CD workflow, `tfinit` and other linters are ran in different containers against the code base way earlier in the workflow than the terraform commands.  Terraform plan and apply was working perfectly as they received the updated folder paths during a refactoring.

**Comment by mgenereu** [comment]
How about a verbosity option to show the files that were inspected?

**Reply by wata727** [reply]
It might not be a bad idea to show the filenames in log. However, the log contains other output as well, so I'm not sure if it can meet your request.
